File: radar/submissions/wrong_answer/radar_ulf.py
#!/usr/bin/env python3
# Solution by Ulf Lundstrom
from sys import stdin

def euclid(a,b):
	if b:
		d,y,x = euclid(b,a%b)
		return d, x, y-a//b*x
	return a, 1, 0

def chinese(a,m,b,n):
	d,x,y = euclid(m,n)
	return (a*y%m*n + b*x%n*m)%(m*n)
	# ret = a*(y+m)%m*n + b*(x+n)%n*m;
	# if ret > m*n:
	# 	ret -= m*n
	# return ret
def chinese3(a,m,b,n,c,o):
	return chinese(chinese(a,m,b,n), m*n, c, o)

m1,m2,m3 = map(int,stdin.readline().split())
x1,x2,x3 = map(int,stdin.readline().split())
y1,y2,y3 = map(int,stdin.readline().split())

best = m1*m2*m3
for d1 in range(-y1,y1+1):
	for d2 in range(-y2,y2+1):
		for d3 in range(-y3,y3+1):
			if (d1 > -y1 and d2 > -y2 and d3 > -y3):
				break
			best = min(best,chinese3((x1+d1)%m1,m1, (x2+d2)%m2,m2,
				(x3+d3)%m3,m3))
print(best)

File: radar/submissions/wrong_answer/radar_ulf.cpp
#include <iostream>
using namespace std;

typedef long long ll;

ll euclid(ll a, ll b, ll &x, ll &y) {
	if (b) {
		ll d = euclid(b, a%b, y, x);
		return y -= a/b*x, d;
	}
	return x=1, y=0, a;
}
ll mod(ll x, ll m) {
	return ((x%m)+m)%m;
}
ll chinese(ll a, ll m, ll b, ll n) {
	ll x,y;
	euclid(m,n,x,y);
	// ll ret = (a*(y+m)%m*n + b*(x+n)%n*m + m*n)%(m*n);
	ll ret = mod(a+m*mod((b-a)*x,n),m*n);
	return ret;
}
ll chinese(ll a, ll m, ll b, ll n, ll c, ll o) {
	return chinese(chinese(a, m, b, n), m*n, c, o);
}

int main() {
	ll m1,m2,m3,x1,x2,x3,y1,y2,y3;
	cin >> m1 >> m2 >> m3 >> x1 >> x2 >> x3 >> y1 >> y2 >> y3;
	ll best = m1*m2*m3;
	for (int d1 = -y1; d1 <= y1; ++d1) {
		for (int d2 = -y2; d2 <= y2; ++d2) {
			for (int d3 = -y3; d3 <= y3; ++d3) {
				if (d3 > -y3 && d1 > -y1 && d2 > -y2)
					break;
				best = min(best, chinese((x1+d1+m1)%m1,m1, (x2+d2+m2)%m2,m2, (x3+d3+m3)%m3,m3));
			}
		}
	}
	cout << best << endl;
}